app/views/activities/_filters.html.erb

```diff
@@ -0,0 +1,29 @@
+<%= form_with url: project_activities_path(current_project), method: :get, remote: true, class: 'd-flex flex-xl-row flex-column gap-3', data: { behavior: 'activity-filters' } do |f| %>
+  <div class="d-flex flex-column flex-sm-row w-100 gap-3">
+    <div class="w-100">
+      <%= f.label :user, class: 'form-label' %>
+      <%= f.select :user, options_from_collection_for_select(users_for_select, :id, :name, params[:user]), { include_blank: 'All users' }, { class: 'form-select' } %>
+    </div>
+    <div class="w-100">
+      <%= f.label :type, class: 'form-label' %>
+      <%= f.select :type, options_from_collection_for_select(current_project.activities.pluck(:trackable_type).uniq, :to_s, :demodulize, params[:type] ), { include_blank: 'All types', selected: params[:type] }, { class: 'form-select' } %>
+    </div>
+  </div>
+
+  <div class="w-100">
+    <span class="form-label mb-2 d-block">Date range</span>
+    <div class="d-flex flex-column flex-sm-row gap-3 w-100">
+      <div class="input-group">
+        <%= f.date_field :period_start, max: params[:period_end] || Date.today, min: "01-01-2016", placeholder: 'From date', value: params[:period_start], class: 'form-control', data: { behavior: 'period-start' } %>
```

Comment:
What's the reasoning behind the min date: `min: "01-01-2016"` ?

Reply:
From BC:

> We should also set a min for the period_start input, currently, we allow dates from year 0001. I'd say a reasonable limit would be Jan 1 2016 (Acitivites were [introduced on Feb 29 2016](https://github.com/dradis/dradis-ce/commit/3368f436b43402b80aad2b16f9ffd82a9df7fb93#diff-6c3192e53b5a2e74e2fe29782dce058fe0398676d09068812b92c40d88625755) so no results would exist before this date).

Rather than Feb 29 I suggested Jan 1 so it's a nice round date, but technically it can be any date before Feb 29 2016.